docs/dev/server-testing.md

```diff
@@ -0,0 +1,91 @@
+# Testing Guidelines for the Server module
```

Comment:
Please correct spelling and grammar mistakes in this file. I've found a handful.